### ISSUE TITLE
Support for filtered positional operator $[<identifier>] - ArrayFilters

### ISSUE THREAD

**Issue by 7robertodantas** [bwaldvogel/mongo-java-server#60]
Hi, it would be great to have support for this operator. I'll see what I can do to help with it in the next few days. 
https://docs.mongodb.com/manual/reference/operator/update/positional-filtered/

**Comment by bwaldvogel** [comment]
I’ve added initial support in 79994ff. Can you test it?

**Comment by 7robertodantas** [comment]
Hi, @bwaldvogel 

It didn't work for my case / scenario.

```
db.services.insertOne({
	"userId": "123",
	"_criteria": [{
		"criteriaName": "CRITERIA_A",
		active: false
	}, {
		"criteriaName": "CRITERIA_B ",
		active: false
	}]
})
```
Then I tried to do the following update (I've tested in the latest version of mongodb and it works in there):

```
db.services.updateOne({ "userId": "123" }, 
{ "$set": { "_criteria.$[elem].active": true } }, 
{
	"arrayFilters": [{
		"elem.criteriaName": {
			"$in": ["CRITERIA_A"]
		}
	}]
})
```

And I got the following error:
```
[js] WriteCommandError: Error parsing array filter :: caused by :: The top-level field name must be an alphanumeric string beginning with a lowercase letter, found 'elem.criteriaName' :
WriteCommandError({
        "ok" : 0,
        "code" : 2,
        "errmsg" : "Error parsing array filter :: caused by :: The top-level field name must be an alphanumeric string beginning with a lowercase letter, found 'elem.criteriaName'"
})
```

**Comment by bwaldvogel** [comment]
Ah right, the validation is too strict for this case.

**Comment by bwaldvogel** [comment]
I’ve improved the support in df0bf7c. Could you retest?

**Comment by 7robertodantas** [comment]
It worked smoothly! Thank you so much @bwaldvogel I think you can close this issue now. :)

**Comment by bwaldvogel** [comment]
I’ve just released version `1.16.0` which contains the changes.
